### ISSUE TITLE
meta-dump creates an error if no metadata is recorded

### ISSUE THREAD

**Issue by christian-monch** [datalad/datalad-metalad#305]
If an entry for a dataset has no metadata recorded, i.e. it has no metadata root record, `meta-dump` fails with:

```
AttributeError: 'NoneType' object has no attribute 'dataset_level_metadata
```

**Comment by christian-monch** [comment]
Fixed by #306

**Comment by christian-monch** [comment]
This is still a problem in tree search:

For example:
```
datalad meta-dump -d https://github.com/christian-monch/metadata-test.git .
```

It also fails locally;
```
> git init test-ds
> cd test-ds
> git fetch https://github.com/christian-monch/metadata-test.git "refs/datalad/*:refs/datalad/*"
> datalad meta-dump .
```
